elastic/kibana
# [Observability Onboarding] Show existing data callout in Firehose flow

Closes https://github.com/elastic/kibana/issues/190795

Adds the logic to display a message to the user in case there is already an existing Firehose data in their cluster and to show the identified AWS services in the "Visualize Data" step right away without waiting for the window to loose focus first.

![CleanShot 2024-12-05 at 11 50 59@2x](https://github.com/user-attachments/assets/00653bf0-f711-4029-9011-a34a160b4b9b)


## How to test

1. Open the Firehose flow
2. Make sure there is no callout and the third step is not active
3. Go to Kibana dev console and ingest some dummy data (see examples bellow)
4. Refresh the page with the Firehose flow
5. make sure there is a callout and the third steps shows the identified AWS service

```
POST logs-aws.apigateway_logs-default/_doc
{
  "@timestamp": "2024-11-25T13:32:01.000Z",
  "some": 111,
  "aws.kinesis.name": "Elastic-CloudwatchLogs"
}

POST metrics-aws.apigateway_metrics-default/_doc
{
    "@timestamp": "2024-11-25T13:31:01.000Z",
    "agent": {
      "type": "firehose"
    },
    "aws": {
      "cloudwatch": {
        "namespace": "AWS/ApiGateway"
      },
      "exporter": {
        "arn": "arn:aws:cloudwatch:us-west-2:975050175126:metric-stream/Elastic-CloudwatchLogsAndMetricsToFirehose-CloudWatchMetricStream-Nhb4NhzPdL4J"
      }
    },
    "cloud": {
      "account": {
        "id": "975050175126"
      },
      "provider": "aws",
      "region": "us-west-2"
    }
}
```
